compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -496,17 +496,25 @@ private void validateEntityFields(Entity entity, NodeList<Node> fields, Module c
                 }
             } else if (typeNode instanceof SimpleNameReferenceNode) {
                 // Verify these are fields annotated with @Relation
+                String attachedEntity = ((SimpleNameReferenceNode) typeNode).name().text();
                 if (field.getRelationAnnotation() != null) {
                     if (this.entities.containsKey(field.getFieldName())) {
                         field.setRelationAttachedToValidEntity(true);
                     } else {
                         // Have to check all entities present in this module
-                        String attachedEntity = ((SimpleNameReferenceNode) typeNode).name().text();
                         validateAttachmentType(entity, field, attachedEntity, typeNode.location(),
                                 currentModule);
                     }
+                } else {
+                    TypeDefinitionNode typeDefinitionNode = getAttachmentEntity(attachedEntity, currentModule);
```

Comment:
There can be issues with this validation in V2 as relation annotation is optional in some cases. Lets fix this once V2 changes are done